### ISSUE TITLE
Beagle return non-zero exist status

### ISSUE THREAD

**Issue by anh151** [sbslee/pypgx#113]
Hello,
I have been able to successfully run PyPGx on many samples. However, for 2 samples I get the error shown below from Beagle. Any ideas of what it could be caused by? Aldy was able to call both samples as *5/*53 and *53/*53. Is there something about the *53 allele that is throwing things off?

```python
def run_pypgx(bam_file, gene="CYP2D6", reference_gene="EGFR", assembly="GRCh38", sample_id=None):
    if sample_id is None:
        sample_id = re.findall(r"\d{7}", bam_file)[0]
    with tempfile.TemporaryDirectory() as tmpdirname:
        vcf = os.path.join(tmpdirname, f"{sample_id}.vcf.gz")
        pypgx.create_input_vcf(
            vcf=vcf,
            fasta=ref_seq,
            bams=bam_file,
            assembly=assembly,
            genes=[gene],
        )
        depth_of_coverage = pypgx.prepare_depth_of_coverage(
            bams=bam_file, assembly=assembly, genes=[gene]
        )
        control_statistics = pypgx.compute_control_statistics(
            gene=reference_gene, bams=bam_file, assembly=assembly
        )
        pypgx.run_ngs_pipeline(
            gene=gene,
            output=tmpdirname,
            variants=vcf,
            depth_of_coverage=depth_of_coverage,
            control_statistics=control_statistics,
            assembly=assembly,
        )
```

```python
Saved VcfFrame[Imported] to: ./imported-variants.zip
Exception in thread "main" java.lang.IllegalArgumentException: 1.0
	at phase.PhaseData.updatePMismatch(PhaseData.java:146)
	at phase.PhaseLS.updateParameters(PhaseLS.java:109)
	at phase.PhaseLS.runStage1(PhaseLS.java:61)
	at main.Main.phaseStage1Variants(Main.java:174)
	at main.Main.phaseAndImpute(Main.java:148)
	at main.Main.main(Main.java:110)

---------------------------------------------------------------------------
CalledProcessError                        Traceback (most recent call last)
/tmp/ipykernel_123/3797058375.py in <module>
     25     depth_of_coverage=depth_of_coverage,
     26     control_statistics=control_statistics,
---> 27     assembly=assembly,
     28 )
     29 #         except ValueError:

~/.local/lib/python3.7/site-packages/pypgx/api/pipeline.py in run_ngs_pipeline(gene, output, variants, depth_of_coverage, control_statistics, platform, assembly, panel, force, samples, exclude, samples_without_sv, do_not_plot_copy_number, do_not_plot_allele_fraction, cnv_caller)
    236         else:
    237             phased_variants = utils.estimate_phase_beagle(
--> 238                 imported_variants, panel=panel)
    239             phased_variants.to_file(f'{output}/phased-variants.zip')
    240             consolidated_variants = utils.create_consolidated_vcf(

~/.local/lib/python3.7/site-packages/pypgx/api/utils.py in estimate_phase_beagle(imported_variants, panel, impute)
    896                 else:
    897                     print(message)
--> 898                     raise e
    899 
    900     return sdk.Archive(metadata, vf3)

~/.local/lib/python3.7/site-packages/pypgx/api/utils.py in estimate_phase_beagle(imported_variants, panel, impute)
    879                     check=True,
    880                     stdout=subprocess.DEVNULL,
--> 881                     stderr=subprocess.PIPE
    882                 )
    883                 vf3 = pyvcf.VcfFrame.from_file(f'{t}/output.vcf.gz')

/opt/conda/lib/python3.7/subprocess.py in run(input, capture_output, timeout, check, *popenargs, **kwargs)
    510         if check and retcode:
    511             raise CalledProcessError(retcode, process.args,
--> 512                                      output=stdout, stderr=stderr)
    513     return CompletedProcess(process.args, retcode, stdout, stderr)
    514 

CalledProcessError: Command '['java', '-Xmx2g', '-jar', '/home/jupyter/.local/lib/python3.7/site-packages/pypgx/api/beagle.22Jul22.46e.jar', 'gt=/tmp/tmpn4rgxgsz/input.vcf', 'chrom=chr22:42116498-42155810', 'ref=/home/jupyter/pypgx-bundle/1kgp/GRCh38/CYP2D6.vcf.gz', 'out=/tmp/tmpn4rgxgsz/output', 'impute=false']' returned non-zero exit status 1.
```

Thanks,
Andrew

**Comment by sbslee** [comment]
@anh151,

Could you send me the input files via email (sbstevenlee@gmail.com)? Since it's a Beagle-related issue, I need to run the pipeline myself in order to figure out what's wrong. I just need the following input files:

```
$ pypgx run-ngs-pipeline \
CYP2D6 \
grch37-CYP2D6-pipeline \
--variants grch37-variants.vcf.gz \
--depth-of-coverage grch37-depth-of-coverage.zip \
--control-statistics grch37-control-statistics-VDR.zip
```

**Comment by sbslee** [comment]
@anh151,

Thanks for sending the files, I was able to reproduce the issue. I'm still figuring out what the error means, but I've never encountered that particular one before, so it may take some time. In the worst case scenario, I will need to contact the author of Beagle. I will get back to you soon!

**Comment by sbslee** [comment]
@anh151,

I have a good news! I contacted the author of Beagle and he said the following:

> There was not enough data for Beagle to estimate model parameters in this analysis, and Beagle's estimated a parameter value that was outside the permitted range..  You should be able to avoid this error by adding the argument "em=false".  In general, haplotype phasing algorithms are most accurate when there are a large number of tightly-linked markers.  Haplotype accuracy will normally be poor if the input data contain only two markers.

So I updated the `0.22.0-dev` branch to set `em=false` and it solved your issue (cf173b0). I sent you the output files via email.

I also did some testing and it seems `em=false` doesn't affect the performance of PyPGx in other normal cases, but please let me know if this update starts to return results that are different from what you originally got for other samples/genes.

I will release the official `0.22.0` version after I make sure everything is working as expected. I would greatly appreciate if you could also test the development version too.
